digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/LogUtils.scala

```diff
@@ -9,24 +9,21 @@ object LogUtils {
 
   /** Logs the time it took function <i>f</i> to complete, when ever it took more than {@link timeLoggingThresholdInMs}.
     * In case <i>f</i> fails, log line is printed as an error, otherwise as debug level line. */
-  def time[R](logger: Logger, operationName: String, noFilter: Boolean = false, url :Option[String] = None)(f: => R): R = {
+  def time[R](logger: Logger, operationName: String, noFilter: Boolean = false, url :Option[String] = None, params: Option[Map[String, String]] = None)(f: => R): R = {
     val begin = System.currentTimeMillis()
+    val paramString = if (params.nonEmpty) s"params = {${params.get.map(a => s"${a._1}: ${a._2}").mkString(", ")}}" else s""
     try {
-      logger.info(s"$operationName started at ${DateTime.now()} ")
+      logger.info(s"$operationName started at ${DateTime.now()}")
       val result    = f
       val duration  = System.currentTimeMillis() - begin
       val urlString = if (url.isDefined) {s"URL: ${url.get}"} else ""
-      if (noFilter) {
-        logger.info(s"$operationName completed in $duration ms and in second ${duration / 1000}, ${urlString}")
-      } else {
-        if (duration >= timeLoggingThresholdInMs) {
-          logger.info(s"$operationName completed in $duration ms and in second ${duration / 1000}, ${urlString}")
-        }
+      if (noFilter || duration >= timeLoggingThresholdInMs) {
+        logger.info(s"$operationName completed in ${duration / 1000}s ${duration%1000}ms. $urlString $paramString")
       }
       result
     } catch {
       case e: Exception =>
-        logger.error(s"$operationName failed.")
+        logger.error(s"$operationName failed. $paramString")
```

Comment:
Lieneekö off topic tähän pullariin, mutta loggauksiin nyt kuitenkin...

Loggautuukos tässä muuten virheen tyyppi/luokka nyt ollenkaan näkyviin, en ole näkevinäni.
Exceptionluokka saattais olla hyödyllistä tietoa. (Vai printataanko se sit vielä erikseen jossain "ylempänä"?)

Reply:
Ei tosiaan printata ja hyvin monessakin API-kutsussa virheestä ei jää sen suurempaa jälkeä lokeihin, kun taas käyttäjälle saattaa palautua stacktrace kokonaisuudessaan

Reply:
Nice, digiroad menee juurti toisin päin. Käyttäjä ei saa mitään tietoa mutta logissa näkyy virhe. 
Ideaalisesti kumpikin viesti pitäisi olla jotenkin linkitetty jollakin id, niin että käyttäjä lähettää virheen id ja sen id avulla kehittäjä löytää virheen.

Sitten on myös UI koodin virheiden logitus joka pitäisi olla ja joka lähettää sen keskitettyyn paikkaan.

Reply:
UUID olis varmaan parempi